crates/cheatcodes/src/utils.rs

```diff
@@ -155,21 +154,20 @@ fn create_wallet(private_key: &U256, label: Option<&str>, state: &mut Cheatcodes
         .abi_encode())
 }
 
-pub(super) fn sign(private_key: &U256, digest: &B256, chain_id: u64) -> Result {
-    let wallet = parse_wallet(private_key)?.with_chain_id(chain_id);
+pub(super) fn sign(private_key: &U256, digest: &B256) -> Result {
+    // The `ecrecover` precompile does not use EIP-155. No chain ID is needed.
+    let wallet = parse_wallet(private_key)?;
 
-    // The `ecrecover` precompile does not use EIP-155
-    let sig = wallet.sign_hash(digest.to_ethers())?;
-    let recovered = sig.recover(digest.to_ethers())?.to_alloy();
+    let sig = wallet.sign_hash_sync(*digest)?;
+    let recovered = sig.recover_address_from_prehash(digest)?;
```

Comment:
so the previous impl also didn't require the chain id?

Reply:
correct